core/http.md

```diff
@@ -0,0 +1,3293 @@
+  to be deleted. However, attributes that are managed by the server might have
+  specialized processing in those cases, in particular, mandatory attributes,
+  as well as ones that have default values defined, MUST be reset to their
+  default values rather than deleted.
+- A `PATCH` operation MUST only modify the attributes explicitly mentioned
+  in the request. Any attribute in the request with a value of `null` MUST be
+  interpreted as a request to delete the attribute, and as with `PUT`/`POST`,
+  server managed attributes might have specialized processing.
+- `PUT` MUST NOT be targeted at xRegistry collections. A `POST` or `PATCH`
+  MUST be used instead to add entities to the collection, and a
+  `DELETE` MUST then be used to delete unwanted entities.
+- Unknown query parameters SHOULD be silently ignored by servers. This
+  includes specification-defined but unsupported query parameters.
+- Despite the
+  [HTTP specification](https://datatracker.ietf.org/doc/html/rfc9110#name-idempotent-methods)
+  saying that the `PUT` method is idempotent, this specification does not
+  adhere to that rule when it comes to the `epoch` and `modifiedat` attributes.
+  While multiple identical `PUT` requests will yield the same semantic effect
+  as single `PUT` for all other attributes, the `epoch` and `modifiedat`
+  attributes are designed to always be updated on each write operation to that
+  entity.
+
+### No-Code Servers
+
+In the [core specification](./spec.md) there is a
+discussion about ["no-code servers"](./spec.md#design-no-code-servers). In the
+case of HTTP, simple file servers SHOULD support exposing Resources where the
+HTTP body response contains the Resource's domain-specific "document" as well
+exposing the serialization of the Resource's xRegistry metadata via the
+`$details` suffix on the URL path. This can be achieved by creating a
+secondary, sibling, file on disk with `$details` at the end of its filename.
+
+## Registry HTTP APIs
+
+This section mainly focuses on the successful interaction patterns of the APIs.
+For example, most examples will show an HTTP "200 OK" as the response. Each
+implementation MAY choose to return a more appropriate response based on the
+specific situation. For example, in the case of an authentication error the
+server could return `401 Unauthorized`.
+
+The following sections define the API in more detail.
+
+### Entity Processing Rules
+
+Rather than repeating the processing rules for `PATCH`, `POST` and `PUT`,
+for each type of xRegistry entity or Registry collection, the overall pattern
+is defined once in this section and any entity-, or collection-specific rules
+will be detailed in the appropriate section in the specification.
+
+#### Creating or Updating Entities
+
+This defines the general rules for how to update entities.
+
+Creating or updating entities MAY be done using HTTP `PUT`, `PATCH` or `POST`
+methods:
+- `PUT    <PATH-TO-ENTITY>                     # Process a single entity
```

Comment:
These are all missing closing backticks

Reply:
got 'em. And found another one elsewhere in the spec - thanks!